index.bs

```diff
@@ -0,0 +1,603 @@
+
+## Permissions ## {#api-permissions}
+
+The <dfn for=PermissionName enum-value>"idle-detection"</dfn> [=powerful
+feature=] is a [=boolean feature=].
+
+## Permissions policy ## {#api-permissions-policy}
+
+This specification defines a [=policy-controlled feature=] identified by the
+string `"idle-detection"`. Its [=default allowlist=] is `'self'`.
+
+<div class="note">
+The [=default allowlist=] of `'self'` allows usage of this feature on
+same-origin nested frames by default but prevents access by third-party content.
+
+Third-party usage can be selectively enabled by adding the
+`allow="idle-detection"` attribute to an {{iframe}} element:
+
+<div class="example">
+```html
+<iframe src="https://example.com" allow="idle-detection"></iframe>
+```
+</div>
+
+Alternatively, this feature can be disabled completely in first-party contexts
+by specifying the permissions policy in an HTTP response header:
+
+<div class="example">
+```http
+Permissions-Policy: idle-detection 'none'
+```
+</div>
+
+See [[PERMISSIONS-POLICY]] for more details.
+</div>
+
+## The {{IdleDetector}} interface ## {#api-idledetector}
+
+<xmp class="idl">
+dictionary IdleOptions {
+  [EnforceRange] unsigned long long threshold;
+  AbortSignal signal;
+};
+
+[
+  SecureContext,
+  Exposed=(Window,DedicatedWorker)
+] interface IdleDetector : EventTarget {
+  constructor();
+  readonly attribute UserIdleState? userState;
+  readonly attribute ScreenIdleState? screenState;
+  attribute EventHandler onchange;
+  [Exposed=Window] static Promise<PermissionState> requestPermission();
+  Promise<void> start(optional IdleOptions options = {});
+};
+</xmp>
+
+Instances of {{IdleDetector}} are created with the [=internal slots=] described
+in the following table:
+
+<table dfn-for=IdleDetector dfn-type=attribute>
+  <tr>
+    <th>[=Internal slot=]</th>
+    <th>Initial value</th>
+    <th>Description (non-normative)</th>
+  </tr>
+  <tr>
+    <td><dfn>\[[state]]</dfn></th>
+    <td>`"stopped"`</td>
+    <td>Tracks the active state of the {{IdleDetector}}</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[threshold]]</dfn></th>
+    <td>`undefined`</td>
+    <td>The configured idle detection threshold</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[userState]]</dfn></td>
+    <td>`null`</td>
+    <td>The last known user idle state</td>
+  </tr>
+  <tr>
+    <td><dfn>\[[screenState]]</dfn></td>
+    <td>`null`</td>
+    <td>The last known screen idle state</td>
+  </tr>
+</table>
+
+<wpt>
+  idlharness-worker.https.window.js
+  idlharness.https.window.js
+</wpt>
+
+Methods on this interface typically complete asynchronously, queuing work on the
+<dfn>idle detection task source</dfn>.
+
+### {{IdleDetector/userState}} attribute ### {#api-idledetector-userstate}
+
+<div algorithm>
+The <dfn attribute for=IdleDetector>userState</dfn> getter steps are:
+
+1.  Return [=this=].{{[[userState]]}}.
+
+</div>
+
+<div algorithm="user becomes idle">
```

Comment:
Should these be nested under the userState attribute section? Maybe they're best in the IdleDetector interface section, or their own dedicated section.

Reply:
I've moved these into their own "Reacting to state changes" section.